src/gt4py/next/program_processors/runners/dace_iterator/utility.py

```diff
@@ -59,11 +60,14 @@ def as_scalar_type(typestr: str) -> ts.ScalarType:
     return ts.ScalarType(kind)
 
 
-def filter_neighbor_tables(offset_provider: dict[str, Any]) -> dict[str, NeighborTable]:
```

Comment:
Before this function only filtered out all values that were not a negbour table.
But now it makes some AST walk walk (which is potentially costly) and then filter them out with some additional condition.
I strongly suggest to you that you change its name to something like `get_used_neigbour_tables()` or something like that to clearly indicate this behaviour.